plutus-ledger-api/test-plugin/Spec/MintValue/V3.hs

```diff
@@ -0,0 +1,153 @@
+
+testPropsInHaskell :: TestTree
+testPropsInHaskell =
+  testGroup
+    "Haskell"
+    [ test_Hask_MintValueBuiltinData
+    , test_Hask_AssetClassIsEitherMintedOrBurned
+    , test_Hask_MintValueMintedIsPositive
+    , test_Hask_MintValueBurnedIsPositive
+    ]
+
+test_Hask_MintValueBuiltinData :: TestTree
+test_Hask_MintValueBuiltinData =
+  testProperty "Builtin data representation of MintValue and Value is the same"
+    $ scaleTestsBy 5 prop_MintValueBuiltinData
+
+test_Hask_AssetClassIsEitherMintedOrBurned :: TestTree
+test_Hask_AssetClassIsEitherMintedOrBurned =
+  testProperty "Asset class is either minted or burned"
+    $ scaleTestsBy 5 prop_AssetClassIsEitherMintedOrBurned
+
+test_Hask_MintValueMintedIsPositive :: TestTree
+test_Hask_MintValueMintedIsPositive =
+  testProperty "MintValue minted is positive"
+    $ scaleTestsBy 5 prop_MintValueMintedIsPositive
+
+test_Hask_MintValueBurnedIsPositive :: TestTree
+test_Hask_MintValueBurnedIsPositive =
+  testProperty "MintValue burned is positive"
+    $ scaleTestsBy 5 prop_MintValueBurnedIsPositive
+
+--------------------------------------------------------------------------------
+-- CEK-evaluated properties ----------------------------------------------------
+
+testPropsInPlinth :: TestTree
+testPropsInPlinth =
+  testGroup
+    "Plinth"
+    [ test_Plinth_MintValueBuiltinData
+    , test_Plinth_AssetClassIsEitherMintedOrBurned
+    , test_Plinth_MintValueMintedIsPositive
+    , test_Plinth_MintValueBurnedIsPositive
+    ]
+
+test_Plinth_MintValueBuiltinData :: TestTree
+test_Plinth_MintValueBuiltinData =
+  testProperty "Builtin data representation of MintValue and Value is the same"
+    $ cekProp
+    . \values ->
+      $$(compile [||prop_MintValueBuiltinData||])
+        `unsafeApplyCode` liftCodeDef values
+
+test_Plinth_AssetClassIsEitherMintedOrBurned :: TestTree
+test_Plinth_AssetClassIsEitherMintedOrBurned =
+  testProperty "Asset class is either minted or burned"
+    $ cekProp
+    . \mintValue ->
+      $$(compile [||prop_AssetClassIsEitherMintedOrBurned||])
+        `unsafeApplyCode` liftCodeDef mintValue
+
+test_Plinth_MintValueMintedIsPositive :: TestTree
+test_Plinth_MintValueMintedIsPositive =
+  testProperty "MintValue minted is positive" $ cekProp . \mintValue ->
+    $$(compile [||prop_MintValueMintedIsPositive||])
+      `unsafeApplyCode` liftCodeDef mintValue
+
+test_Plinth_MintValueBurnedIsPositive :: TestTree
+test_Plinth_MintValueBurnedIsPositive =
+  testProperty "MintValue burned is positive" $ cekProp . \mintValue ->
+    $$(compile [||prop_MintValueBurnedIsPositive||])
+      `unsafeApplyCode` liftCodeDef mintValue
+
+--------------------------------------------------------------------------------
+-- Helper functions ------------------------------------------------------------
+
+scaleTestsBy :: (QC.Testable prop) => Haskell.Int -> prop -> QC.Property
+scaleTestsBy factor =
+  QC.withMaxSuccess (100 Haskell.* factor) . QC.mapSize (Haskell.* factor)
```

Comment:
I guess we should move that to some common `QuickCheck` module. But we can do that in another PR.